src/games/strategy/triplea/ui/AbstractMovePanel.java

```diff
@@ -138,11 +143,67 @@ public final String undoMove(final int moveIndex) {
     return undoMove(moveIndex, false);
   }
 
+  /**
+   * Executes an undo move for any of the units passed in as a parameter.
+   *
+   * "Cannot undo" Error messages are suppressed if any moves cannot be undone
+   * (at least until we come up with a way to deal with "n" reasons for an undo
+   * failure rather than just one)
+   */
+  public void undoMoves(Set<Unit> units) {
+    Set<UndoableMove> movesToUndo = getMovesToUndo(units, getMoveDelegate().getMovesMade());
+
+    if (movesToUndo.size() == 0) {
+      String error = "Could not undo any moves, check that the unit has moved and that you can undo the move normally";
+      JOptionPane.showMessageDialog(getTopLevelAncestor(), error, "Could not undo move", JOptionPane.ERROR_MESSAGE);
+      return;
+    }
+
+    undoMovesInReverseOrder(movesToUndo);
+  }
+
+  private static Set<UndoableMove> getMovesToUndo(Set<Unit> units, List<Object> movesMade) {
+    Set<UndoableMove> movesToUndo = Sets.newHashSet();
+
+    if (movesMade != null) {
```

Comment:
Not currently, but a public method calls into this method which could allow for the value to be null.